catchup/ledgerFetcher_test.go

```diff
@@ -67,38 +135,42 @@ func TestLedgerFetcherErrorResponseHandling(t *testing.T) {
 	}
 	listener, err := net.Listen("tcp", "localhost:")
 
+	var httpServerResponse = 0
+	var contentTypes = make([]string, 0)
 	require.NoError(t, err)
 	go s.Serve(listener)
 	defer s.Close()
 	defer listener.Close()
-
-	httpServerResponse := http.StatusNotFound
-	contentTypes := make([]string, 0)
 	mux.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
 		for _, contentType := range contentTypes {
 			w.Header().Add("Content-Type", contentType)
 		}
 		w.WriteHeader(httpServerResponse)
 	})
-
+	successPeer := testHTTPPeer(listener.Addr().String())
 	lf := makeLedgerFetcher(&mocks.MockNetwork{}, &mocks.MockCatchpointCatchupAccessor{}, logging.TestingLog(t), &dummyLedgerFetcherReporter{}, config.GetDefaultLocal())
-	peer := testHTTPPeer(listener.Addr().String())
-	err = lf.getPeerLedger(context.Background(), &peer, basics.Round(0))
-	require.Equal(t, errNoLedgerForRound, err)
 
-	httpServerResponse = http.StatusInternalServerError
-	err = lf.getPeerLedger(context.Background(), &peer, basics.Round(0))
-	require.Equal(t, fmt.Errorf("getPeerLedger error response status code %d", httpServerResponse), err)
+	// headLedger non-http peer
+	err = lf.headLedger(context.Background(), nil, basics.Round(0))
+	require.Equal(t, errNonHTTPPeer, err)
 
-	httpServerResponse = http.StatusOK
-	err = lf.getPeerLedger(context.Background(), &peer, basics.Round(0))
-	require.Equal(t, fmt.Errorf("getPeerLedger : http ledger fetcher invalid content type count %d", 0), err)
+	// headLedger parseURL failure
+	parseFailurePeer := testHTTPPeer("foobar")
+	err = lf.headLedger(context.Background(), &parseFailurePeer, basics.Round(0))
+	require.Equal(t, fmt.Errorf("could not parse a host from url"), err)
+
+	// headLedger 404 response
+	httpServerResponse = http.StatusNotFound
```

Comment:
I don't think you need this anymore?

Reply:
This is the response that the mux handler func writes on behalf of the peer. These tests ensure the behavior of the `headLedger` function is correct based on various possible peer responses.

Reply:
yes but in the error construction you are not using this httpServerResponse variable, you are using the http package directly.